pkg/controller/networkpolicy/clustergroup.go

```diff
@@ -198,5 +197,26 @@ func (n *NetworkPolicyController) syncInternalGroup(key string) error {
 	}
 	klog.V(2).Infof("Updating existing internal Group %s with %d GroupMembers", key, len(memberSet))
 	n.internalGroupStore.Update(updatedGrp)
+	// If a ClusterGroup is added/updated, it might have a reference in ClusterNetworkPolicy.
+	// Trigger the sync for corresponding groups with the shared key and complete the
+	// corresponding group if not yet completed.
+	n.addressGroupMutex.Lock()
+	addrGroupObj, found, _ := n.addressGroupStore.Get(key)
+	if found {
+		// If AddressGroup was created corresponding to this ClusterGroup, complete the
+		// AddressGroup fields before syncing the AddressGroup.
+		addrGroup := addrGroupObj.(*antreatypes.AddressGroup)
+		if isGroupSelectorUnset(addrGroup.Selector) {
```

Comment:
Maybe I missed something but what if addrGroup.Selector is set, but goes out of sync due to CG update? `updateClusterGroup` does not enqueue addressGroups. Shouldn't we also check addrGroup.Selector == updatedGrp.Selector here?

Reply:
that's correct .. will update

Reply:
done